### PR TITLE
[MM-19227] shift focus onto the next settings item

### PR DESCRIPTION
**Pull request by Willyfrog**
#### Summary
Once you save one of item from user account settings, focus should go to the next section so the user can edit everything in one go.

also fixed .eslintrc.json wasn't configured for jsx within .jsx files
also fixed `codeBlockOnCtrlEnter` was receiving a string while expecting a boolean.

#### Ticket Link

[MM-19227](https://mattermost.atlassian.net/browse/MM-19227)

### CODE REVIEW

File: components/user_settings/advanced/user_settings_advanced.jsx [hunk 1]
```diff
@@ -85,6 +86,11 @@ export default class AdvancedSettingsDisplay extends React.PureComponent {
         };
     }
 
+    componentDidMount() {
+        const prev = this.isDeactivateEnabled() ? 'deactivateAccount' : 'advancedPreviewFeatures';
```
Comment (jwilander) [inline components/user_settings/advanced/user_settings_advanced.jsx]: Worth throwing `this.isDeactivateEnabled() ? 'deactivateAccount' : 'advancedPreviewFeatures'` in its own method? I see it being used in one other place

**Comment by esethna** [comment]
This PR seems to break

 keyboard navigation in the Account Settings. Devin should review.

It seems you can no longer hit ENTER on an edit field and have the first field focused in the expanded setting.

See community-release:
settings

**Comment by Willyfrog** [comment]
Added setting focus to the maximized setting.

- moved maximized setting to /components, as it now uses the store to comunicate it is closing.
- fixed tests as that broke many parts
- fixed a couple of warnings appearing during compiling

**Comment by Willyfrog** [comment]
@jwilander @thekiiingbob I think you are both making valid points, but this change has gone too big, should we move those to a separate ticket so we can refactor the settings as every tab is different from the previous one?

File: components/user_settings/advanced/user_settings_advanced.jsx (hunk 1 above)
Comment (Willyfrog) [reply components/user_settings/advanced/user_settings_advanced.jsx]: just trying to make explicit the intent, I can replace with its contents if you prefer that way.

**Comment by esethna** [comment]
The first focused item doesn't seem to be visually focused, even though it is. Below "Desktop Notifications" is seemingly skipped by the visual focus, but I can still tab to it. Same happens after saving a setting now

tabbing

**Comment by esethna** [comment]
Is this PR necessary? Seems like we get the expected behaviour from this PR? ie you can keep tabbing through additional settings after hitting save.

https://github.com/mattermost/mattermost-webapp/pull/3946

File: components/user_settings/advanced/user_settings_advanced.jsx (hunk 1 above)
Comment (jwilander) [reply components/user_settings/advanced/user_settings_advanced.jsx]: 0/5, I'll leave it up to you

**Comment by jwilander** [comment]
@Willyfrog My only concern about separate tickets is that they will get lost in the backlog and only completed after we run into a bunch of bugs like this. I think this is a case where we should be focusing on high quality over rushing something in. I like @thekiiingbob's suggestion of a cypress test that would be able to check if something breaks. With that I will be much happier merging this as is

**Comment by Willyfrog** [comment]
Thanks @jwilander ! It's a good point, and I'll try hard to make sure it doesn't fall into oblivion. But if we start a refactor on this section, I think we should do it of the whole `Account Settings` section, as every part is a bit different from the other and we should standardize to make sure we don't run into easy to break code, which is what created this bug originally (it's a regression from 5.10). Unfortunately, it is a big enough task that I don't think I should take it on my current sprint (this task already grew way too big with 47 files changed) and having it grow bigger will definitely generate bigger problems. Anyways I went ahead and created a ticket for [the whole refactor](https://mattermost.atlassian.net/browse/MM-19576) with a special note to ensure that we don't miss your suggestion :)

On the other hand, I'll do bob's suggestion of a cypress test, as that will help both now and when we do the refactor, ensuring less errors.

**Comment by jwilander** [comment]
Sounds good to me 👍

**Comment by Willyfrog** [comment]
@esethna I added @devinbinnie changes for the keyboard navigation

**Comment by esethna** [comment]
Thanks @Willyfrog, can you help get the test server back up? ^

**Comment by Willyfrog** [comment]
added focus for radio input

File: e2e/cypress/integration/account_settings/general/focus_spec.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+// - Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
+describe('Account Settings > General', () => {
+    before(() => {
+        // # Go to Account Settings with "user-1"
+        cy.toAccountSettingsModal('user-1');
+    });
+
+    it('focus is in editing the full name', () => {
```
Comment (thekiiingbob) [inline e2e/cypress/integration/account_settings/general/focus_spec.js]: The name of the test is the same across all the files. They should be different, right?

**Comment by esethna** [comment]
@Willyfrog i think this piece of feedback may have been missed in the iteration? 
image

**Comment by Willyfrog** [comment]
@esethna I've fixed 1 and 2, I'm not really sure I understand the third point, can we discuss in a zoom call?

File: e2e/cypress/integration/account_settings/general/focus_spec.js (hunk 2 above)
Comment (Willyfrog) [reply e2e/cypress/integration/account_settings/general/focus_spec.js]: since they are in different folders, I thought it would be ok. I'll change them

**Comment by esethna** [comment]
@Willyfrog should we close this PR and reassign/reopen the ticket or is it still in progress?

**Comment by Willyfrog** [comment]
@esethna since these parts are all moved to ts, we have to recreate the whole thing again. Closing, no need to create new tickets, the original one is still valid, but we need to port most of this work to the new language

**Pull request closed without merging**